### ISSUE TITLE
"Operation was cancelled" error

### ISSUE THREAD

**Issue by paldepind** [simolus3/moor#2818]
I'm trying to delete some rows that break a foreign key constraint.

My code is doing the following: In the `beforeOpen` function for `MigrationStrategy` I'm finding all the offending rows using `PRAGMA foreign_key_check` and then attempting to delete the rows in a for loop. The first delete works fine, but the second delete throws the error below. If I restart my app another delete will go through, and the second will trigger the error again.

```
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: Operation was cancelled
#0      DriftCommunication.request (package:drift/src/remote/communication.dart:113:66)
#1      _RemoteQueryExecutor.ensureOpen (package:drift/src/remote/client_impl.dart:157:10)
#2      LazyDatabase.ensureOpen.<anonymous closure> (package:drift/src/utils/lazy_database.dart:61:49)
#3      _rootRunUnary (dart:async/zone.dart:1407:47)
#4      _CustomZone.runUnary (dart:async/zone.dart:1308:19)
<asynchronous suspension>
#5      DatabaseConnectionUser.doWhenOpened.<anonymous closure> (package:drift/src/runtime/api/connection_user.dart:162:55)
<asynchronous suspension>
#6      NonNullableCancellationExtension.resultOrNullIfCancelled (package:drift/src/runtime/cancellation_zone.dart:62:24)
<asynchronous suspension>
#7      QueryStream.fetchAndEmitData (package:drift/src/runtime/executor/stream_queries.dart:314:20)
<asynchronous suspension>
```

I don't know what the error is supposed to mean and can't find anything in the documentation about cancelling operations. But, it seems that the first delete is doing something that causes any subsequent query to fail with that error (I've tried with select queries and they give the same error).

Here is the full code:

```dart
  MigrationStrategy get migration {
    return MigrationStrategy(
        // ...
        beforeOpen: (_) async {
          if (kDebugMode) {
            final wrongForeignKeys = await customSelect('PRAGMA foreign_key_check').get();
            for (final e in wrongForeignKeys) {
              if (e.data['table'] == 'sessions_tags') {
                final id = e.data['rowid'] as int;
                final n = await (delete(sessionsTags)..where((row) => row.rowId.equals(id))).go();
                // The next line prints once and the following delete fails
                print('Cleaned up violation ($n)');
              }
            }
          }
        });
  }
```

Let me know if there's more information that is useful to debug this.

**Comment by simolus3** [comment]
Can you try to use  [an explicit drift isolate](https://drift.simonbinder.eu/docs/advanced-features/isolates/#manually-managing-drift-isolates) and pass `isolateDebugLog: true` to `connect()`? Having those logs would be very useful to narrow this down.

The `QueryStream.fetchAndEmitData` in the stack trace seems suspicious, is it possible that there is a stream query being listened to at that point?

**Comment by paldepind** [comment]
Thanks a lot for the response. I really appreciate it.

> The `QueryStream.fetchAndEmitData` in the stack trace seems suspicious, is it possible that there is a stream query being listened to at that point?

My app uses stream queries that execute when the app is opened. But I'm surprised if they seem to be running concurrently with the code in `beforeOpen` as I would expect that to run before any queries.

> Can you try to use [an explicit drift isolate](https://drift.simonbinder.eu/docs/advanced-features/isolates/#manually-managing-drift-isolates) and pass `isolateDebugLog: true` to `connect()`? Having those logs would be very useful to narrow this down.

I tried to do that. It made the error go away and all the deletes went through immediately. Below are the changes that I made to my code, based on the docs that you linked to. Of course it's nice that the deletes no go through, but still, I would have expected the original code to work as well.

```diff
 class MyDatabase extends _$MyDatabase {
-  MyDatabase() : super(_openConnection());
+  MyDatabase([QueryExecutor? executor]) : super(executor ?? _openConnection());
```

```diff
-  var db = MyDatabase();
+  final db = MyDatabase(
+    DatabaseConnection.delayed(Future.sync(() async {
+      final isolate = await createIsolateWithSpawn();
+      return isolate.connect(singleClientMode: true, isolateDebugLog: true);
+    })),
+  );
```

```diff
+Future<DriftIsolate> createIsolateWithSpawn() async {
+  final token = RootIsolateToken.instance!;
+  return await DriftIsolate.spawn(() {
+    BackgroundIsolateBinaryMessenger.ensureInitialized(token);

+    return LazyDatabase(() async {
+      final dbFolder = await getApplicationDocumentsDirectory();
+      final file = File(p.join(dbFolder.path, 'my-app-database.sqlite'));
+      return NativeDatabase(file);
+    });
+  });
+}
```

Here is the log output I got after `isolateDebugLog: true`. But since the error did not occur with this I suspect that the log is not terribly useful.

<details>
<summary>Log Output</summary>
```
Restarted application in 2,991ms.
flutter: [IN]: Request (id = 0): ServerInfo(SqlDialect.sqlite)
flutter: [OUT]: SuccessResponse (id = 0): null
flutter: [OUT]: Request (id = 0): EnsureOpen(1, null)
flutter: [IN]: Request (id = 1): RunBeforeOpen(Instance of 'OpeningDetails', 0)
flutter: [OUT]: Request (id = 1): EnsureOpen(1, 0)
flutter: [IN]: SuccessResponse (id = 1): true
flutter: [OUT]: Request (id = 2): StatementMethod.select: PRAGMA foreign_key_check with [] (@0)
flutter: [IN]: SuccessResponse (id = 2): Instance of 'SelectResult'
flutter: {table: sessions_tags, rowid: 5628, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 3): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5628] (@0)
flutter: [IN]: SuccessResponse (id = 3): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5629, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 4): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5629] (@0)
flutter: [IN]: SuccessResponse (id = 4): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5630, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 5): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5630] (@0)
flutter: [IN]: SuccessResponse (id = 5): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5631, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 6): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5631] (@0)
flutter: [IN]: SuccessResponse (id = 6): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5632, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 7): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5632] (@0)
flutter: [IN]: SuccessResponse (id = 7): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5633, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 8): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5633] (@0)
flutter: [IN]: SuccessResponse (id = 8): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5634, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 9): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5634] (@0)
flutter: [IN]: SuccessResponse (id = 9): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5635, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 10): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5635] (@0)
flutter: [IN]: SuccessResponse (id = 10): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5636, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 11): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5636] (@0)
flutter: [IN]: SuccessResponse (id = 11): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5637, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 12): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5637] (@0)
flutter: [IN]: SuccessResponse (id = 12): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5638, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 13): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5638] (@0)
flutter: [IN]: SuccessResponse (id = 13): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5639, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 14): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5639] (@0)
flutter: [IN]: SuccessResponse (id = 14): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5640, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 15): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5640] (@0)
flutter: [IN]: SuccessResponse (id = 15): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5641, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 16): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5641] (@0)
flutter: [IN]: SuccessResponse (id = 16): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5642, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 17): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5642] (@0)
flutter: [IN]: SuccessResponse (id = 17): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5643, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 18): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5643] (@0)
flutter: [IN]: SuccessResponse (id = 18): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5644, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 19): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5644] (@0)
flutter: [IN]: SuccessResponse (id = 19): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5645, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 20): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5645] (@0)
flutter: [IN]: SuccessResponse (id = 20): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5646, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 21): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5646] (@0)
flutter: [IN]: SuccessResponse (id = 21): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5647, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 22): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5647] (@0)
flutter: [IN]: SuccessResponse (id = 22): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5648, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 23): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5648] (@0)
flutter: [IN]: SuccessResponse (id = 23): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5649, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 24): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5649] (@0)
flutter: [IN]: SuccessResponse (id = 24): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5650, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 25): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5650] (@0)
flutter: [IN]: SuccessResponse (id = 25): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5651, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 26): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5651] (@0)
flutter: [IN]: SuccessResponse (id = 26): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5652, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 27): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5652] (@0)
flutter: [IN]: SuccessResponse (id = 27): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5653, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 28): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5653] (@0)
flutter: [IN]: SuccessResponse (id = 28): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5654, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 29): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5654] (@0)
flutter: [IN]: SuccessResponse (id = 29): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5655, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 30): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5655] (@0)
flutter: [IN]: SuccessResponse (id = 30): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5656, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 31): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5656] (@0)
flutter: [IN]: SuccessResponse (id = 31): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5657, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 32): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5657] (@0)
flutter: [IN]: SuccessResponse (id = 32): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5658, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 33): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5658] (@0)
flutter: [IN]: SuccessResponse (id = 33): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5659, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 34): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5659] (@0)
flutter: [IN]: SuccessResponse (id = 34): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5660, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 35): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5660] (@0)
flutter: [IN]: SuccessResponse (id = 35): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5661, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 36): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5661] (@0)
flutter: [IN]: SuccessResponse (id = 36): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5664, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 37): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5664] (@0)
flutter: [IN]: SuccessResponse (id = 37): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5665, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 38): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5665] (@0)
flutter: [IN]: SuccessResponse (id = 38): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5666, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 39): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5666] (@0)
flutter: [IN]: SuccessResponse (id = 39): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5667, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 40): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5667] (@0)
flutter: [IN]: SuccessResponse (id = 40): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5668, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 41): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5668] (@0)
flutter: [IN]: SuccessResponse (id = 41): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5669, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 42): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5669] (@0)
flutter: [IN]: SuccessResponse (id = 42): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5670, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 43): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5670] (@0)
flutter: [IN]: SuccessResponse (id = 43): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5671, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 44): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5671] (@0)
flutter: [IN]: SuccessResponse (id = 44): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5672, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 45): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5672] (@0)
flutter: [IN]: SuccessResponse (id = 45): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5673, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 46): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5673] (@0)
flutter: [IN]: SuccessResponse (id = 46): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5674, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 47): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5674] (@0)
flutter: [IN]: SuccessResponse (id = 47): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5675, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 48): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5675] (@0)
flutter: [IN]: SuccessResponse (id = 48): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5676, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 49): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5676] (@0)
flutter: [IN]: SuccessResponse (id = 49): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5677, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 50): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5677] (@0)
flutter: [IN]: SuccessResponse (id = 50): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5678, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 51): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5678] (@0)
flutter: [IN]: SuccessResponse (id = 51): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5679, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 52): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5679] (@0)
flutter: [IN]: SuccessResponse (id = 52): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5680, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 53): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5680] (@0)
flutter: [IN]: SuccessResponse (id = 53): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5681, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 54): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5681] (@0)
flutter: [IN]: SuccessResponse (id = 54): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5682, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 55): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5682] (@0)
flutter: [IN]: SuccessResponse (id = 55): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5683, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 56): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5683] (@0)
flutter: [IN]: SuccessResponse (id = 56): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5684, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 57): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5684] (@0)
flutter: [IN]: SuccessResponse (id = 57): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5685, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 58): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5685] (@0)
flutter: [IN]: SuccessResponse (id = 58): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5686, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 59): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5686] (@0)
flutter: [IN]: SuccessResponse (id = 59): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5687, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 60): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5687] (@0)
flutter: [IN]: SuccessResponse (id = 60): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5688, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 61): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5688] (@0)
flutter: [IN]: SuccessResponse (id = 61): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5689, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 62): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5689] (@0)
flutter: [IN]: SuccessResponse (id = 62): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5690, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 63): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5690] (@0)
flutter: [IN]: SuccessResponse (id = 63): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5691, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 64): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5691] (@0)
flutter: [IN]: SuccessResponse (id = 64): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5692, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 65): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5692] (@0)
flutter: [IN]: SuccessResponse (id = 65): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5693, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 66): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5693] (@0)
flutter: [IN]: SuccessResponse (id = 66): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5694, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 67): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5694] (@0)
flutter: [IN]: SuccessResponse (id = 67): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5695, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 68): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5695] (@0)
flutter: [IN]: SuccessResponse (id = 68): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5696, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 69): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5696] (@0)
flutter: [IN]: SuccessResponse (id = 69): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5697, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 70): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5697] (@0)
flutter: [IN]: SuccessResponse (id = 70): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5698, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 71): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5698] (@0)
flutter: [IN]: SuccessResponse (id = 71): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5699, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 72): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5699] (@0)
flutter: [IN]: SuccessResponse (id = 72): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5700, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 73): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5700] (@0)
flutter: [IN]: SuccessResponse (id = 73): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5701, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 74): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5701] (@0)
flutter: [IN]: SuccessResponse (id = 74): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5702, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 75): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5702] (@0)
flutter: [IN]: SuccessResponse (id = 75): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5703, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 76): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5703] (@0)
flutter: [IN]: SuccessResponse (id = 76): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5704, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 77): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5704] (@0)
flutter: [IN]: SuccessResponse (id = 77): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5705, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 78): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5705] (@0)
flutter: [IN]: SuccessResponse (id = 78): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5706, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 79): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5706] (@0)
flutter: [IN]: SuccessResponse (id = 79): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5707, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 80): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5707] (@0)
flutter: [IN]: SuccessResponse (id = 80): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5708, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 81): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5708] (@0)
flutter: [IN]: SuccessResponse (id = 81): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5709, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 82): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5709] (@0)
flutter: [IN]: SuccessResponse (id = 82): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5710, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 83): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5710] (@0)
flutter: [IN]: SuccessResponse (id = 83): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5711, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 84): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5711] (@0)
flutter: [IN]: SuccessResponse (id = 84): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5712, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 85): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5712] (@0)
flutter: [IN]: SuccessResponse (id = 85): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5713, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 86): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5713] (@0)
flutter: [IN]: SuccessResponse (id = 86): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5714, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 87): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5714] (@0)
flutter: [IN]: SuccessResponse (id = 87): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5715, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 88): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5715] (@0)
flutter: [IN]: SuccessResponse (id = 88): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5716, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 89): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5716] (@0)
flutter: [IN]: SuccessResponse (id = 89): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5717, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 90): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5717] (@0)
flutter: [IN]: SuccessResponse (id = 90): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5718, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 91): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5718] (@0)
flutter: [IN]: SuccessResponse (id = 91): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5719, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 92): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5719] (@0)
flutter: [IN]: SuccessResponse (id = 92): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5720, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 93): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5720] (@0)
flutter: [IN]: SuccessResponse (id = 93): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5721, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 94): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5721] (@0)
flutter: [IN]: SuccessResponse (id = 94): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5722, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 95): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5722] (@0)
flutter: [IN]: SuccessResponse (id = 95): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5723, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 96): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5723] (@0)
flutter: [IN]: SuccessResponse (id = 96): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5724, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 97): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5724] (@0)
flutter: [IN]: SuccessResponse (id = 97): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5725, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 98): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5725] (@0)
flutter: [IN]: SuccessResponse (id = 98): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5726, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 99): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5726] (@0)
flutter: [IN]: SuccessResponse (id = 99): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5727, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 100): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5727] (@0)
flutter: [IN]: SuccessResponse (id = 100): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5728, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 101): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5728] (@0)
flutter: [IN]: SuccessResponse (id = 101): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5729, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 102): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5729] (@0)
flutter: [IN]: SuccessResponse (id = 102): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5730, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 103): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5730] (@0)
flutter: [IN]: SuccessResponse (id = 103): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5731, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 104): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5731] (@0)
flutter: [IN]: SuccessResponse (id = 104): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5732, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 105): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5732] (@0)
flutter: [IN]: SuccessResponse (id = 105): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5733, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 106): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5733] (@0)
flutter: [IN]: SuccessResponse (id = 106): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5734, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 107): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5734] (@0)
flutter: [IN]: SuccessResponse (id = 107): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5735, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 108): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5735] (@0)
flutter: [IN]: SuccessResponse (id = 108): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5736, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 109): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5736] (@0)
flutter: [IN]: SuccessResponse (id = 109): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5737, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 110): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5737] (@0)
flutter: [IN]: SuccessResponse (id = 110): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5738, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 111): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5738] (@0)
flutter: [IN]: SuccessResponse (id = 111): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5739, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 112): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5739] (@0)
flutter: [IN]: SuccessResponse (id = 112): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5740, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 113): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5740] (@0)
flutter: [IN]: SuccessResponse (id = 113): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5741, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 114): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5741] (@0)
flutter: [IN]: SuccessResponse (id = 114): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5742, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 115): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5742] (@0)
flutter: [IN]: SuccessResponse (id = 115): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5743, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 116): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5743] (@0)
flutter: [IN]: SuccessResponse (id = 116): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5744, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 117): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5744] (@0)
flutter: [IN]: SuccessResponse (id = 117): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5745, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 118): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5745] (@0)
flutter: [IN]: SuccessResponse (id = 118): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5746, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 119): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5746] (@0)
flutter: [IN]: SuccessResponse (id = 119): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5747, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 120): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5747] (@0)
flutter: [IN]: SuccessResponse (id = 120): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5748, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 121): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5748] (@0)
flutter: [IN]: SuccessResponse (id = 121): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5749, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 122): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5749] (@0)
flutter: [IN]: SuccessResponse (id = 122): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5750, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 123): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5750] (@0)
flutter: [IN]: SuccessResponse (id = 123): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5751, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 124): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5751] (@0)
flutter: [IN]: SuccessResponse (id = 124): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5752, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 125): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5752] (@0)
flutter: [IN]: SuccessResponse (id = 125): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5753, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 126): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5753] (@0)
flutter: [IN]: SuccessResponse (id = 126): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5754, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 127): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5754] (@0)
flutter: [IN]: SuccessResponse (id = 127): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5755, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 128): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5755] (@0)
flutter: [IN]: SuccessResponse (id = 128): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5756, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 129): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5756] (@0)
flutter: [IN]: SuccessResponse (id = 129): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5757, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 130): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5757] (@0)
flutter: [IN]: SuccessResponse (id = 130): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5758, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 131): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5758] (@0)
flutter: [IN]: SuccessResponse (id = 131): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5759, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 132): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5759] (@0)
flutter: [IN]: SuccessResponse (id = 132): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5760, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 133): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5760] (@0)
flutter: [IN]: SuccessResponse (id = 133): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5761, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 134): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5761] (@0)
flutter: [IN]: SuccessResponse (id = 134): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5762, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 135): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5762] (@0)
flutter: [IN]: SuccessResponse (id = 135): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5763, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 136): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5763] (@0)
flutter: [IN]: SuccessResponse (id = 136): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5764, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 137): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5764] (@0)
flutter: [IN]: SuccessResponse (id = 137): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5765, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 138): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5765] (@0)
flutter: [IN]: SuccessResponse (id = 138): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5766, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 139): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5766] (@0)
flutter: [IN]: SuccessResponse (id = 139): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5767, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 140): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5767] (@0)
flutter: [IN]: SuccessResponse (id = 140): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5768, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 141): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5768] (@0)
flutter: [IN]: SuccessResponse (id = 141): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5769, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 142): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5769] (@0)
flutter: [IN]: SuccessResponse (id = 142): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5770, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 143): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5770] (@0)
flutter: [IN]: SuccessResponse (id = 143): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5771, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 144): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5771] (@0)
flutter: [IN]: SuccessResponse (id = 144): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5772, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 145): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5772] (@0)
flutter: [IN]: SuccessResponse (id = 145): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5773, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 146): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5773] (@0)
flutter: [IN]: SuccessResponse (id = 146): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5774, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 147): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5774] (@0)
flutter: [IN]: SuccessResponse (id = 147): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5775, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 148): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5775] (@0)
flutter: [IN]: SuccessResponse (id = 148): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5776, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 149): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5776] (@0)
flutter: [IN]: SuccessResponse (id = 149): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5777, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 150): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5777] (@0)
flutter: [IN]: SuccessResponse (id = 150): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5778, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 151): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5778] (@0)
flutter: [IN]: SuccessResponse (id = 151): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5779, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 152): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5779] (@0)
flutter: [IN]: SuccessResponse (id = 152): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5780, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 153): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5780] (@0)
flutter: [IN]: SuccessResponse (id = 153): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5781, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 154): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5781] (@0)
flutter: [IN]: SuccessResponse (id = 154): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5782, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 155): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5782] (@0)
flutter: [IN]: SuccessResponse (id = 155): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5783, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 156): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5783] (@0)
flutter: [IN]: SuccessResponse (id = 156): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5784, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 157): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5784] (@0)
flutter: [IN]: SuccessResponse (id = 157): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5785, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 158): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5785] (@0)
flutter: [IN]: SuccessResponse (id = 158): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5791, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 159): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5791] (@0)
flutter: [IN]: SuccessResponse (id = 159): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5792, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 160): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5792] (@0)
flutter: [IN]: SuccessResponse (id = 160): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5793, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 161): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5793] (@0)
flutter: [IN]: SuccessResponse (id = 161): 1
flutter: Cleaned up violation (1)
flutter: {table: sessions_tags, rowid: 5794, parent: sessions, fkid: 0}
flutter: [OUT]: Request (id = 162): StatementMethod.deleteOrUpdate: DELETE FROM "sessions_tags" WHERE "_rowid_" = ?; with [5794] (@0)
flutter: [IN]: SuccessResponse (id = 162): 1
flutter: Cleaned up violation (1)
flutter: [OUT]: Request (id = 163): StatementMethod.custom: PRAGMA foreign_keys = ON with [] (@0)
flutter: [IN]: SuccessResponse (id = 163): null
flutter: [OUT]: SuccessResponse (id = 1): null
flutter: [IN]: SuccessResponse (id = 0): true
flutter: [OUT]: Request (id = 164): StatementMethod.select: SELECT * FROM "sessions" WHERE "duration" IS NULL; with []
flutter: [OUT]: Request (id = 165): StatementMethod.select: SELECT * FROM "tags"; with []
flutter: [OUT]: Request (id = 166): StatementMethod.select: SELECT "sessions"."created_at" AS "sessions.created_at", "sessions"."changed_at" AS "sessions.changed_at", "sessions"."synced_at" AS "sessions.synced_at", "sessions"."id" AS "sessions.id", "sessions"."goal_duration" AS "sessions.goal_duration", "ses
flutter: sions"."started_at" AS "sessions.started_at", "sessions"."ended_at" AS "sessions.ended_at", "sessions"."utc_offset" AS "sessions.utc_offset", "sessions"."duration" AS "sessions.duration", "sessions"."pause_segments" AS "sessions.pause_segments", "sessions_tags"."tag_id" AS "sessions_tags.tag_id", "s
flutter: essions_tags"."session_id" AS "sessions_tags.session_id" FROM "sessions" LEFT OUTER JOIN "sessions_tags" ON "sessions_tags"."session_id" = "sessions"."id" WHERE "sessions"."started_at" >= ? ORDER BY "sessions"."started_at" DESC; with [1701039600]
flutter: [IN]: SuccessResponse (id = 164): Instance of 'SelectResult'
flutter: [IN]: SuccessResponse (id = 165): Instance of 'SelectResult'
flutter: [IN]: SuccessResponse (id = 166): Instance of 'SelectResult'
```

</details>

**Comment by simolus3** [comment]
Thanks for providing the details! But the log output looks just like what I would expect it to look like, nothing stands out unfortunately.

> I would have expected the original code to work as well.

Yeah, me too. And `_openConnection()` has just been using path provider + `NativeDatabase.createInBackground` before or has there been some custom setup in there?

**Comment by paldepind** [comment]
Yes, my `_openConnection` is very normal I think:

```dart
LazyDatabase _openConnection() {
  // the LazyDatabase util lets us find the right location for the file async.
  return LazyDatabase(() async {
    // put the database file, called db.sqlite here, into the documents folder
    // for your app.
    final dbFolder = await getApplicationDocumentsDirectory();
    final file = File(p.join(dbFolder.path, 'deep-time-database.sqlite'));
    return NativeDatabase.createInBackground(file);
  });
}
```

Based on your comment above I removed a stream query that is initiated on the first page of my app. With this query removed all the deletes (the original code without explicit isolate creation) go through without errors. So it seems to be related to the stream query running in parallel with the code in `beforeOpen`. Is guess this should not happen as queries should wait till the `beforeOpen` is done?

**Comment by simolus3** [comment]
I think I'm beginning to form a theory for this, but I couldn't reproduce this yet. 

> Is guess this should not happen as queries should wait till the `beforeOpen` is done?

Definitely! Stream queries have a mechanism to cancel pending work it goes stale or if there aren't any subscriptions left. I wonder if this could have happened because listening to the stream is what caused the database to be opened, but then the subscription got cancelled before the database is fully open. That still shouldn't interfere with other statements though.

If you bring the stream query back, but do this immediately after constructing your database, does the error still happen?

```dart
var db = MyDatabase();
db.customSelect('select 1').get();
```

You don't even need to await that, it would just be interesting to know what happens when that runs before the stream is listened to.

**Comment by paldepind** [comment]
Hi again. Sorry for the delayed response, I somehow missed your reply.

> If you bring the stream query back, but do this immediately after constructing your database, does the error still happen?

No, that fixes the error! With `db.customSelect('select 1').get();` right after construction of `MyDatabase` everything works flawlessly and if I remove it again the error returns.

**Comment by imtoori** [comment]
Hey this is happening for us too! 
The fix works, but it would be nice to have this fixed :)

**Comment by simolus3** [comment]
I agree - but I still couldn't find a way to reproduce this locally. I think I've also made improvements on this in b3e2aa739fc41b69560936584234f1022ad8c6ab which my be related. Which drift version are you using?

**Comment by FMorschel** [comment]
I've also made some other changes to my full app (migrations and some logic changes), but anyway, I was on 2.18.0 and now on 2.20.0 I do not need the `db.customSelect('select 1').get();` workaround anymore.

**Comment by imtoori** [comment]
We're on 2.18.0 at the moment, I'll see if we can update

**Comment by FMorschel** [comment]
Update:

We did need the above-mentioned workaround again, but only for a new table in our DB. Not sure what it has that is different from all of the others in this case. But this time I got no `Operation was cancelled` message, only the delete would not complete. Took me some time to remember that this issue was related to deletes. Noticed when setting the `isolateDebugLog` fixed it.

**Comment by Tim-Otte** [comment]
Hello everyone and a happy new year!

It seems that I have the same issue that you guys had. I am working with a Stream that queries data from the database and Displays the info on the first screen of my app. Additionally I have a query that runs when I build the basic layout of my app to add default data into the database once the basic layout is rendered. Apparently the query interferes with the select query stream and I get an "Operation was cancelled" error, but the funny part is that I only get the error in my widget test.

I am using Flutter 3.27.1, drift 2.20.2 and drift_flutter 0.2.0.

The error:
```
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following CancellationException was thrown running a test:
Operation was cancelled

When the exception was thrown, this was the stack:
#0      checkIfCancelled (package:drift/src/runtime/cancellation_zone.dart:94:5)
#1      _BaseExecutor._synchronized.<anonymous closure> (package:drift/src/runtime/executor/helpers/engines.dart:56:9)
#3      Lock.synchronized.callBlockAndComplete (package:drift/src/utils/synchronized.dart:17:21)
#4      Lock.synchronized.<anonymous closure> (package:drift/src/utils/synchronized.dart:21:35)
<asynchronous suspension>
<asynchronous suspension>
(elided 7 frames from dart:async and package:stack_trace)
```

The insert query:
```dart
/// Creates default serving sizes in the database.
///
/// This method inserts a list of default serving sizes into the database.
/// The serving sizes include grams, milliliters, ounces, and fluid ounces.
///
/// Parameters:
/// - [localizations]: An instance of `AppLocalizations` used to get localized names for the serving sizes.
///
/// Returns a [Future] that completes when the serving sizes have been inserted.
Future<void> createDefaultServingSizes(AppLocalizations localizations) async {
  await servingSize.insertAll([
    ServingSizeCompanion.insert(
      id: const Value(1),
      name: localizations.serving_size_gram,
      short: const Value('g'),
      measuringUnit: MeasurementUnit.metric,
      isLiquid: const Value(false),
      valueInBaseServingSize: 1,
    ),
    ServingSizeCompanion.insert(
      id: const Value(2),
      name: localizations.serving_size_milliliter,
      short: const Value('ml'),
      measuringUnit: MeasurementUnit.metric,
      isLiquid: const Value(true),
      valueInBaseServingSize: 1,
    ),
    // … more rows to insert
  ], mode: InsertMode.insertOrIgnore);
}
```

The select query:
```dart
/// Calculates the total calories, carbs, fats, and proteins consumed for today.
///
/// This method retrieves the consumption data for the current day and computes
/// the total amount of calories, carbs, fats, and proteins consumed.
///
/// Returns a [Stream] that emits a [Nutriments] object containing
/// the total calories, carbs, fats, and proteins consumed for today.
Stream<Nutriments> calculateTotalNutrimentsForToday({
  int? withoutConsumptionId,
}) {
  final today = DateTime.now();
  final startOfDay = DateTime(today.year, today.month, today.day);
  final endOfDay = startOfDay.add(Duration(days: 1));

  final caloriesInConsumption = managers.consumption.computedField((c) =>
      c.quantity *
      c.productCode.caloriesPer100Units /
      Constant(100.0) *
      c.servingSizeId.valueInBaseServingSize);

  final carbsInConsumption = managers.consumption.computedField((c) =>
      c.quantity *
      c.productCode.carbsPer100Units /
      Constant(100.0) *
      c.servingSizeId.valueInBaseServingSize);

  final fatsInConsumption = managers.consumption.computedField((c) =>
      c.quantity *
      c.productCode.fatPer100Units /
      Constant(100.0) *
      c.servingSizeId.valueInBaseServingSize);

  final proteinsInConsumption = managers.consumption.computedField((c) =>
      c.quantity *
      c.productCode.proteinsPer100Units /
      Constant(100.0) *
      c.servingSizeId.valueInBaseServingSize);

  final manager = managers.consumption.withFields([
    caloriesInConsumption,
    carbsInConsumption,
    fatsInConsumption,
    proteinsInConsumption,
  ]);

  var query =
      manager.filter((f) => f.loggedOn.isBetween(startOfDay, endOfDay));

  if (withoutConsumptionId != null) {
    query = query.filter((f) => f.id.equals(withoutConsumptionId).not());
  }

  return query.watch().map(
        (data) => data.fold<Nutriments>(
          Nutriments.empty,
          (a, b) => Nutriments(
            calories: a.calories +
                (caloriesInConsumption.read(b.$2)?.toDouble() ?? 0),
            carbs: a.carbs + (carbsInConsumption.read(b.$2)?.toDouble() ?? 0),
            fats: a.fats + (fatsInConsumption.read(b.$2)?.toDouble() ?? 0),
            proteins: a.proteins +
                (proteinsInConsumption.read(b.$2)?.toDouble() ?? 0),
          ),
        ),
      );
}
```

If you have any idea, how I could fix this or improve the code, let me know. If you want to reproduce the error, the repo of this code is the calorie-counter repo in my profile.

**Comment by simolus3** [comment]
Excellent catch @Tim-Otte, thank you! I'm not sure this is the same error that others were seeing here, but I managed to find multiple drift bugs with your test cases.

The problem here was that the manager API uses multiple queries in a transaction to resolve one-to-many relations (as it should!). All queries in drift are supposed to support cancellations: When the source data changes and we're in the middle of running multiple queries, we already know that we'll throw the results away and don't have to bother with the remaining queries.

There's a bug when opening transactions in a context that was cancelled causing the unhandled exception you've seen here. I've fixed that in 0901c984a987e54ba0b67e227adbfdcf3c08eeb4, and I'll release the fix this weekend. @Tim-Otte, you can try using a [development version of drift](https://drift.simonbinder.eu/Internals/#using-an-unreleased-drift-version) until then. I was at least able to run your tests independently, but I guess there is a missing `tearDown()` to make a simple `flutter test` pass? Let me know if you need help with the tests, I owe you one for that awesome testcase :sweat_smile:

**Comment by Tim-Otte** [comment]
Hey @simolus3, thank you for your fast response and the bugfix. I am happy my project helped you find some bugs 😄 

I am still trying to figure out the Flutter tests (I usually use .NET and Flutter is very different in some behaviours 🥲) and did run into a weird behaviour with drift and the `tearDownAll` method. I was following the [documentation](https://drift.simonbinder.eu/testing/#writing-tests) and closed the database connection, but then the whole test process ran forever. I had to use the Task Manager to stop it from running because I could not stop the test from the terminal or from inside VS Code. When I removed the `tearDownAll`, the tests ran fine, as you can see [here](https://github.com/Tim-Otte/calorie-counter/actions/runs/12549620431/job/34991010780). But it seems that this was also fixed with your fix. I will also test it with your development version :+1:

Thanks again for your fast help and kind response. I owe you one for being able to use your library in my project. 😊

P.S.: If you (as a Flutter / Dart expert) know a good way to improve [this](https://github.com/Tim-Otte/calorie-counter/blob/ef07ec134aba3ac90f857ef41117935380f97d82/lib/src/layout.dart#L29) (creating default data while being able to access the localization), I would be very grateful. But this not very important, only if you have time 😊
